tests/lib/cbprintf_package/src/main.c

```diff
@@ -923,7 +923,9 @@ ZTEST(cbprintf_package, test_cbprintf_package_convert_static)
 	uint32_t copy_flags = CBPRINTF_PACKAGE_CONVERT_RW_STR;
 
 	clen = cbprintf_package_convert(spackage, slen, NULL, 0, copy_flags, NULL, 0);
-	zassert_true(clen == slen + sizeof(test_str1) + 1/*null*/ - 2 /* arg+ro idx gone*/);
+	zassert_true(clen >= 0, "cbprintf_package_convert failed: %d", clen);
+	zassert_true((unsigned int)clen ==
+		     slen + sizeof(test_str1) + 1 /*null*/ - 2 /* arg+ro idx gone*/);
```

Comment:
I'm a bit perplexed as to why this `unsinfed int` cast is necessary. Both `clen` and `slen` are `int`. Perhaps the following might be better?

```suggestion
	zassert_true(clen ==
		     slen + (int)sizeof(test_str1) + 1 /*null*/ - 2 /* arg+ro idx gone*/);
```

Reply:
Indeed. Fixed.